labs/a2.md

```diff
@@ -0,0 +1,186 @@
+
+Now we will create a simplistic package using the hellopenguin executable that you will make in the coming steps. First, move into the a2 folder in the repository that you cloned in the Getting Started section:
+
+`cd decal-labs/a2`
+
+Now we are going to create a folder to work in for this exercise:
+
+`mkdir ex1`
+
+And now move into the folder:
+
+`cd ex1`
+
+### Writing and Compiling the Program
+
+Now, we will make a very simple application in C that prints “Hello Penguin!” named hellopenguin. Invoke:
+
+`touch hellopenguin.c`
+
+This will create an empty file named `hellopenguin.c`. Now, using the a preferred text editor of your choice, such as `vim`, `emacs`, or `nano`, insert the following code into `hellopenguin.c`
+
+~~~~
+#include <stdio.h>
+
+int main()
+
+{
+
+   printf("Hello Penguin!\n");
+
+   return 0;
+
+}
+~~~~
+
+
+We will now compile the source file that you have just written:
+
+`gcc hellopenguin.c -o hellopenguin`
+
+What this does is to take in a source file `hellopenguin.c` and compile it to an executable named `hellopenguin` with the `-o` output flag.
+
+### Packaging the executable
+
+Now, we will create the folder structure of where the executable shall reside in. In Debian, user-level packages usually reside in the folder `/usr/bin/`:
+
+`mkdir -p packpenguin/usr/bin`
+
+Now move your compiled `hellopenguin` exectuable into the `packpenguin/usr/bin/` folder.
+
+`mv hellopenguin packpenguin/usr/bin/`
+
+Now we will create a package called `hellopenguin`. Move into the parent directory of the `hellopenguin` folder and invoke the following:
+
+`fpm -s dir -t deb -n hellopenguin -v 1.0~ocfdecal1 -C packpenguin`
+
+This specifies that you want to take in a directory, using the `-s` flag, and to output a `.deb` package using the `-t` flag. It takes in a directory called `packpenguin`, using the `-C` flag, and output a `.deb` file named `hellopenguin`, using the `-n`, with a version number of `1.0~ocf1`, using the `-v` flag.
+
+Now test it by invoking apt and installing it:
+
+`sudo dpkg -i ./hellopenguin_1.0~ocf1_amd64.deb`
+
+Now you should be able to run `hellopenguin` by doing the following:
+
+`hellopenguin`
+
+### Exercise 2: Troubleshooting
+Now we are going to try and troubleshoot a package. Move to the other folder, `ex2`.
+
+Try installing the `ocfspy` package using `dpkg`. It should error. Take note what it is erroring on! Now try and fix it.
+
+Hint: Inspect the package for more details. The file to create that application is in the folder. Try compiling and packaging it. Exercise 1 may be a useful reference if you are stuck.
+
+### Conclusion
+After you’re done, complete [this](https://goo.gl/forms/WMhdZXAcTDPG7gOW2) submission form.
```

Comment:
Please add the feedback questions at the end of the lab.

Reply:
Feedback questions are on page 2 of the submission form